lib/blocks.php

```diff
@@ -142,25 +142,37 @@ function get_dynamic_blocks_regex() {
 function gutenberg_render_block( $block ) {
 	global $post;
 
-	$block_type    = WP_Block_Type_Registry::get_instance()->get_registered( $block['blockName'] );
-	$is_dynamic    = $block['blockName'] && null !== $block_type && $block_type->is_dynamic();
+	$global_post = $post;
+	$pre_render  = apply_filters( 'block_pre_render', $block );
```

Comment:
All filters must be documented:

https://make.wordpress.org/core/handbook/best-practices/inline-documentation-standards/php/#4-hooks-actions-and-filters

Which would be particularly helpful here because it's not clear to me the purpose in testing for `null`, or what the use-case is for `pre_render`.

I might have seen it as something of an inverse where a non-`null` filtered value would be inferred as the overridden value, similar to what's done in `WP_oEmbed::get_html`:

https://github.com/WordPress/WordPress/blob/da7a80d67fea29c2badfc538bfc01c8a585f0cbe/wp-includes/class-oembed.php#L375-L394

Reply:
thanks @aduth - I'm not ready to document them because I'm not sure what I would document. trying to figure out the interface first 😉 

I'll make the description in the PR clearer

Reply:
Hah, I was just thinking about something along the same line as a solution for #11704.

As @aduth mentioned, the usual method would be to call the filter `pre_render_block`, and to pass it a `null` value as the first argument. `$block` would then be passed as the second argument.

If the `pre_render_block` rendered a non-`null` value (including an empty string, as #11704 would do), `render_block()` would return that value, instead of running the rest of the function.

Reply:
documented!